src/store/AssetInformationStore.js

```diff
@@ -18,30 +18,69 @@ export default class AssetInformationStore {
 
     onContextReady = () => {
         this.stx.prepend('headsUpHR', this.update);
-        this.stx.prepend('createXAxis', this.update);
     };
 
-    @action.bound update() {
-        if (!this.visible) { return; }
+    _container = null;
+    _spot = null;
+    _open = null;
+    _close = null;
+    _high = null;
+    _low  = null;
+
+
+    update = () => {
+        // do not query dom on each frame
+        if (!this._container) {
+            this._container = this.context.topNode.querySelector('.ciq-asset-information');
+
+            this._spot = this._container.querySelector('.ciq-ai-spot');
+            this._open = this._container.querySelector('.ciq-ai-open');
+            this._close = this._container.querySelector('.ciq-ai-close');
+            this._high = this._container.querySelector('.ciq-ai-high');
+            this._low = this._container.querySelector('.ciq-ai-low');
+        }
+        const container = this._container;
+
+        if (!this.visible) {
+            container.style.visibility = 'hidden';
+            return;
+        }
 
         const bar = this.stx.barFromPixel(this.stx.cx);
         const prices = this.stx.chart.xaxis[bar];
 
-        if (!prices) { return; }
+        if (!prices) {
+            return;
+        }
+
 
         const {
             Open, High, Low, Close,
         } = prices.data || { };
-        this.open = (Open && !this.isTick) ? Open.toFixed(this.decimalPlaces) : null;
-        this.high = (High && !this.isTick) ? High.toFixed(this.decimalPlaces) : null;
-        this.low = (Low && !this.isTick) ? Low.toFixed(this.decimalPlaces) : null;
-        this.close = (Close && !this.isTick) ? Close.toFixed(this.decimalPlaces) : null;
-        this.price = Close ? Close.toFixed(this.decimalPlaces) : null;
-    }
 
-    @observable price = null;
-    @observable open = null;
-    @observable high = null;
-    @observable low = null;
-    @observable close = null;
+        if (!Open && !High && !Low && !Close) {
+            container.style.visibility = 'hidden';
+            return;
+        }
+
+        container.style.visibility = 'visible';
+
+        if (this.isTick) {
+            const children = container.children;
+            const visibility = this.isTick ? 'hidden' : 'visible';
+
+            for (let idx = 1; idx < children.length; ++idx) {
+                children[idx].style.visibility = visibility;
+            }
+        }
+
+
+        if (this._container) {
+            this._open.textContent = (Open && !this.isTick) ? Open.toFixed(this.decimalPlaces) : '';
```

Comment:
I see we are manipulating DOM directly, is this better than passing props, why breaking component to smaller components is not an option?

Reply:
@nazaninreihani  some stx.prepend() and stx.append() callbacks are triggered inside animation loop. 

direct DOM manipulation is an anti-pattern, but we need it here because every 1ms counts.  
Even a smaller component has to pay for REACT reconciler & Mobx observable tracking.
And some of these callbacks where already the smallest component you could get.

In this PR we are knowingly choosing to manipulate DOM in the animation frames for the sake of performance.